packages/core/src/dispatcher.ts

```diff
@@ -171,8 +162,8 @@ export class Dispatcher {
     // TODO Add validation the message adheres to the proper format.
 
     const { doc: docId, id } = message
-    if (await this.repository.has(docId)) {
-      const document = await this.repository.get(docId)
+    const document = await this.repository.get(docId)
+    if (document) {
```

Comment:
Does this actually load the document into memory if we reply to a query?
Seems like it should be possible to just retrieve the tip without loading the document into memory?

Reply:
It is possible. And, I see the point in doing that.

Reply:
done, btw.